### ISSUE TITLE
RUF045 on dataclass method assignment

### ISSUE THREAD

**Issue by beskep** [astral-sh/ruff#16360]
### Description

ruff version: 0.9.7
command: `ruff check test.py --select RUF --preview`

```python
import functools
from collections.abc import Callable
from dataclasses import dataclass


@dataclass
class Vector:
    x: float
    y: float

    def __call__(self) -> None:
        print(self)

    # RUF045: Assignment without annotation found in dataclass body
    call = __call__

    call2: Callable[..., None] = __call__  # this makes new field

    def add(self, x: float, y: float):
        return Vector(x=self.x + x, y=self.y + y)

    # RUF045: Assignment without annotation found in dataclass body
    add_x = functools.partialmethod(add, y=0)
```

I am not sure if it's intended, but ruff check RUF045 for assigning a method of dataclass.

Is there a more appropriate way for typing new method, perhaps?

**Comment by beskep** [comment]
`add_x: ClassVar = functools.partialmethod(add, y=0)` worked fine.
I had not thought of typing `ClassVar` for a function. How about adding this case to the example?

Or, perhaps, how about applying RUF045 only before `def` section?
It would increase false negatives though.

**Comment by beskep** [comment]
I appreciate RUF045 (I forget to type dataclass fields time to time), but `ClassVar[Callable[..., None]]` and `ClassVar[partialmethod[Vector]]` seems a bit verbose and cumbersome to me.
I'd rather mark noqa for those lines.

**Comment by CommentatorForAll** [comment]
I do agree with OP here, from my understanding, partialmethod returns a function that requires `self`. Hence the assigned variable is a callable on the object not the class. Why does one need to annotate it as a classvar then?

**Comment by ntBre** [comment]
I ended up agreeing with Dylan and moved ahead with stabilizing the rule in 0.16, so I'm closing this as an answered question.
